### PR TITLE
Update deferred deeplinking iOS

### PR DESCRIPTION
**Pull request by IloveJavaa**
Asana - https://app.asana.com/0/0/1204212631804152/f

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,12 +12,13 @@ let module_trackier_emitter = null;
...
     module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
 } else if (Platform.OS === "ios") {
-    //module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
+    module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
 }
 
 var TrackierConfig = function(appToken,environment) {
 	this.appToken = appToken;
 	this.environment = environment;
+	this.appSecret = {}
```
Comment (Hemant-Mann) [inline index.js:21]: let's add 2 props instead of 1 @IloveJavaa 
this.secretId = 
this.secretKey =

Comment (Hemant-Mann) [reply index.js:21]: also we'll need to update the index.d.ts file

File: ios/RNTrackierSDK.swift [hunk 2]
```diff
@@ -9,25 +9,26 @@ import Foundation
...
+		config.setAppSecret(secretId: appSecret["secretId"] as! String, secretKey: appSecret["secretKey"] as! String)
 		config.setSDKVersion(sdkVersion: "1.6.39")
-		config.setAppSecret(secretId: secretId, secretKey: secretKey)
+		if(deeplinking == true){
+			config.setDeeplinkListerner(listener: self)
+		}
 		TrackierSDK.initialize(config: config)
 	}
 
-	 @objc func setAppSecret(_ dict: NSDictionary) -> Void {
```
Comment (Hemant-Mann) [inline ios/RNTrackierSDK.swift:27]: if we are deprecating this function then let's also remove the function from TrackierSDK

File: index.js (hunk 1 above)
Comment (IloveJavaa) [reply index.js]: All changes done

File: ios/RNTrackierSDK.swift [hunk 3]
```diff
@@ -9,25 +9,26 @@ import Foundation
...
 		let appToken = dict["appToken"] as! String;
 		let environment = dict["environment"] as! String;
+		let deeplinking = dict["hasDeferredDeeplinkCallback"] as! Bool?
 		let config = TrackierSDKConfig(appToken: appToken , env: environment)
 		config.setSDKType(sdkType: "react_native_sdk")
+		let appSecret:Dictionary<String,Any> = dict["appSecret"] as? Dictionary<String,Any> ?? [:]
+		config.setAppSecret(secretId: appSecret["secretId"] as! String, secretKey: appSecret["secretKey"] as! String)
 		config.setSDKVersion(sdkVersion: "1.6.39")
-		config.setAppSecret(secretId: secretId, secretKey: secretKey)
+		if(deeplinking == true){
```
[the remark this answers is not in the document]
Comment (IloveJavaa) [reply ios/RNTrackierSDK.swift]: done

File: ios/RNTrackierSDK.swift (hunk 2 above)
Comment (IloveJavaa) [reply ios/RNTrackierSDK.swift:27]: ok

File: index.js [hunk 4]
```diff
@@ -10,14 +10,16 @@ const module_trackier = NativeModules.TrackierSDK;
...
+	module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
 } else if (Platform.OS === "ios") {
-    //module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
+	module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
 }
 
 var TrackierConfig = function(appToken,environment) {
 	this.appToken = appToken;
 	this.environment = environment;
+	this.secretId = {}
```
Comment (Hemant-Mann) [inline index.js:21]:
```suggestion
	this.secretId = '';
```

File: index.js [hunk 5]
```diff
@@ -10,14 +10,16 @@ const module_trackier = NativeModules.TrackierSDK;
...
+	this.secretKey = {}
```
Comment (Hemant-Mann) [inline index.js:22]:
```suggestion
	this.secretKey = '';
```

File: ios/RNTrackierSDK.swift [hunk 6]
```diff
@@ -9,25 +9,26 @@ import Foundation
...
+		<#code#>
+	}
 
 	@objc func initializeSDK(_ dict: NSDictionary) -> Void {
 		let appToken = dict["appToken"] as! String;
 		let environment = dict["environment"] as! String;
+		let deeplinking = dict["hasDeferredDeeplinkCallback"] as! Bool?
 		let config = TrackierSDKConfig(appToken: appToken , env: environment)
 		config.setSDKType(sdkType: "react_native_sdk")
+		let appSecret:Dictionary<String,Any> = dict["appSecret"] as? Dictionary<String,Any> ?? [:]
```
Comment (Hemant-Mann) [inline ios/RNTrackierSDK.swift:23]: this is a string value

Comment (Hemant-Mann) [reply ios/RNTrackierSDK.swift:23]: plz fix it

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]

File: ios/RNTrackierSDK.swift (hunk 6 above)
Comment (IloveJavaa) [reply ios/RNTrackierSDK.swift:23]: done

File: index.js (hunk 5 above)
Comment (IloveJavaa) [reply index.js:22]: done

File: index.js (hunk 4 above)
Comment (IloveJavaa) [reply index.js:21]: done

File: index.js [hunk 7]
```diff
@@ -26,8 +28,9 @@ TrackierSDK.initialize = function(trackierConfig) {
 	module_trackier.initializeSDK(trackierConfig)
 }
 
-TrackierSDK.setAppSecret = function(secretId, secretKey){
-	module_trackier.setAppSecret({"secretId": secretId, "secretKey": secretKey})
+TrackierConfig.prototype.setAppSecret = function(key, value){
```
Comment (Hemant-Mann) [inline index.js:31]:
```suggestion
TrackierConfig.prototype.setAppSecret = function(key, value) {
```

File: index.js [hunk 8]
```diff
@@ -10,14 +10,16 @@ const module_trackier = NativeModules.TrackierSDK;
...
+	module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
 } else if (Platform.OS === "ios") {
-    //module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
+	module_trackier_emitter = new NativeEventEmitter(NativeModules.TrackierSDK);
 }
 
 var TrackierConfig = function(appToken,environment) {
 	this.appToken = appToken;
 	this.environment = environment;
+	this.secretId = ''
```
Comment (Hemant-Mann) [inline index.js:21]:
```suggestion
	this.secretId = '';
```

File: index.js [hunk 9]
```diff
@@ -10,14 +10,16 @@ const module_trackier = NativeModules.TrackierSDK;
...
+	this.secretKey = ''
```
Comment (Hemant-Mann) [inline index.js:22]:
```suggestion
	this.secretKey = '';
```

File: ios/RNTrackierSDK.swift [hunk 10]
```diff
@@ -9,25 +9,25 @@ import Foundation
...
 	@objc func initializeSDK(_ dict: NSDictionary) -> Void {
 		let appToken = dict["appToken"] as! String;
 		let environment = dict["environment"] as! String;
+		let deeplinking = dict["hasDeferredDeeplinkCallback"] as! Bool?
 		let config = TrackierSDKConfig(appToken: appToken , env: environment)
 		config.setSDKType(sdkType: "react_native_sdk")
+		config.setAppSecret(secretId: dict["secretId"] as! String, secretKey: dict["secretKey"] as! String)
 		config.setSDKVersion(sdkVersion: "1.6.39")
-		config.setAppSecret(secretId: secretId, secretKey: secretKey)
+		if((deeplinking) != nil) {
```
Comment (Hemant-Mann) [inline ios/RNTrackierSDK.swift:25]:
```suggestion
		if (deeplinking != nil) {
```

**Review by Hemant-Mann — APPROVED** [review]

File: index.js [hunk 11]
```diff
@@ -95,8 +98,13 @@ TrackierConfig.EnvironmentProduction = "production";
 TrackierConfig.EnvironmentTesting = "testing";
 
 TrackierConfig.prototype.setDeferredDeeplinkCallbackListener = function(deferredDeeplinkCallbackListener) {
-	this.hasDeferredDeeplinkCallback = true;
-	module_trackier_emitter.addListener('trackier_deferredDeeplink', deferredDeeplinkCallbackListener);
+	if (Platform.OS === "android") {
```
Comment (Hemant-Mann) [inline index.js:101]: if else block can be removed because code is same

**Review by Hemant-Mann — APPROVED** [review]

**Pull request merged**